packages/jsdoc-plugins/tests/relation-fixer/addmissingdoclets.js

```diff
@@ -0,0 +1,82 @@
+/**
+ * @license Copyright (c) 2016-2017, CKSource - Frederico Knabben. All rights reserved.
+ * Licensed under the terms of the MIT License (see LICENSE.md).
+ */
+
+/* jshint mocha:true */
+
+'use strict';
+
+const chai = require( 'chai' );
+const expect = chai.expect;
+const addMissingDoclets = require( '../../lib/relation-fixer/addmissingdoclets' );
+const cloneDeep = require( 'lodash' ).cloneDeep;
+
+describe( 'Plugin adds missing doclets from relation chain', () => {
+	it( 'should add missing doclets coming from interfaces', () => {
+		const testDoclets = require( './test-data/interface' );
```

Comment:
This is why the functions with side-effects or modifying the input are so hard to debug.